blog/2015/09/thinking-about-teaching.html

```diff
@@ -0,0 +1,360 @@
+---
+layout: blog
+root: ../../..
+author: Greg Wilson
+title: "Thinking About Teaching"
+date: 2015-09-25
+time: "02:00:00"
+category: ["Teaching"]
+---
+<!-- start excerpt -->
+<p>
+  A little over a year ago,
+  we blogged about <a href="{{page.root}}/blog/2014/09/building-better-teachers.html">jugyokenkyu</a>,
+  or "lesson study",
+  a bucket of practices that Japanese teachers use to hone their craft,
+  from observing each other at work
+  to discussing the lesson afterward
+  to studying curriculum materials with colleagues.
+  Getting the Software Carpentry Foundation up and running almost immediately pushed that aside,
+  but now that the SCF is up and running,
+  it's time to return to the subject.
+  Discussion of <a href="how-teaching-knowledge-is-transferred.html">how teaching practices are transferred</a>
+  is part of that;
+  so are two other developments this week.
+</p>
+<!-- end excerpt -->
+<p>
+  First, Victor Eijkhout has been thinking about
+  <a href="https://bitbucket.org/VictorEijkhout/parallel-computing-book/downloads/eijkhout-teachmpi.pdf">how to teach MPI</a>.
+  In particular,
+  he has been thinking about how the order in which concepts are usually introduced can mislead or confuse,
+  and about a different order that leads more naturally to proficient use.
+  We're not going to include MPI in Software Carpentry any time soon (if ever),
+  but I hope that the next major revision of our lessons will include this kind of design rationale.
+</p>
+<p>
+  Second,
+  <a href="{{page.root}}/pages/team.html#bostroem.a">Azalee Bostroem</a> surveyed people who have taught our current Python lesson
+  to find out what they're actually doing and how it has been going.
+  <a href="python-lesson-survey-results.pdf">The results</a> are fascinating&mdash;here
+  are some of the key findings, and my comments:
+</p>
+<dl>
+  <dt>
+    60% of our lessons are being taught to novices.
+  </dt>
+  <dd>
+    <p>
+      The term "novice" is a bit ambiguous here:
+      it can mean either "someone who has never programmed before"
+      or "someone who hasn't programmed <em>in Python</em>".
+      When we re-design this lesson,
+      we could well wind up creating different versions for these audiences,
+      since someone who's learning Python as a second language will be able to move faster and farther
+      than someone who is new to ideas like "variable", "list", and "function".
+    </p>
+  </dd>
+  <dt>
+    Roughly half of the respondents don't like the way we jump right
+    into an example.
+  </dt>
+  <dd>
+    <p>
+      The intention was to get to an authentic task as quickly as possible,
+      and "load CSV, create plot" is something every scientist does.
+      I still think it works for people who are learning Python as a second language,
+      but if that's only 1/3 of our workshops,
+      it's not surprising that it often doesn't work.
```

Comment:
I wouldn't equate preferring to have an introductions with the current approach not working. I have taught it that way and it does work, but maybe not as an intended quick peak. This might be a good place to mention the number of people who do start with 01-numpy are going through it with a lot of explanation (2nd to last question)